framework/src/meshgenerators/OverlayMeshGenerator.C

```diff
@@ -0,0 +1,138 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#include "OverlayMeshGenerator.h"
+#include "CastUniquePointer.h"
+// libMesh includes
+#include "libmesh/mesh_generation.h"
+#include "libmesh/unstructured_mesh.h"
+#include "libmesh/replicated_mesh.h"
+#include "libmesh/mesh_modification.h"
+#include "libmesh/mesh_tools.h"
+
+registerMooseObject("MooseApp", OverlayMeshGenerator);
+InputParameters
+OverlayMeshGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+  params.addClassDescription("This OverlayMeshGenerator creates a Cartesian mesh using "
+                             "DistributedRectilinearMeshGenerator in "
+                             "the mesh block region.");
+  params.addRequiredParam<MeshGeneratorName>("input", "The base mesh we want to overlay");
+  MooseEnum dims("1=1 2 3");
+  params.addRequiredParam<MooseEnum>(
+      "dim", dims, "The dimension of the mesh to be generated"); // Make this parameter required
+  params.addParam<dof_id_type>("nx", 1, "Number of elements in the X direction");
+  params.addParam<dof_id_type>("ny", 1, "Number of elements in the Y direction");
+  params.addParam<dof_id_type>("nz", 1, "Number of elements in the Z direction");
+  params.addParam<processor_id_type>(
+      "num_cores_for_partition",
+      0,
+      "Number of cores for partitioning the graph (dafaults to the number of MPI ranks)");
+  params.addRangeCheckedParam<unsigned>(
+      "num_side_layers",
+      2,
+      "num_side_layers>=1 & num_side_layers<5",
+      "Number of layers of off-processor side neighbors is reserved during mesh generation");
+  MooseEnum partition("graph linear square", "graph", false);
+  params.addParam<MooseEnum>(
+      "partition", partition, "Which method (graph linear square) use to partition mesh");
+  MooseEnum elem_types = MooseMesh::elemTypes();
+  params.addParam<MooseEnum>("elem_type",
+                             elem_types,
+                             "The type of element from libMesh to "
+                             "generate (default: linear element for "
+                             "requested dimension)");
+
+  params.addRangeCheckedParam<Real>(
+      "bias_x",
+      1.,
+      "bias_x>=0.5 & bias_x<=2",
+      "The amount by which to grow (or shrink) the cells in the x-direction.");
+  params.addRangeCheckedParam<Real>(
+      "bias_y",
+      1.,
+      "bias_y>=0.5 & bias_y<=2",
+      "The amount by which to grow (or shrink) the cells in the y-direction.");
+  params.addRangeCheckedParam<Real>(
+      "bias_z",
+      1.,
+      "bias_z>=0.5 & bias_z<=2",
+      "The amount by which to grow (or shrink) the cells in the z-direction.");
```

Comment:
since DistributedMeshGenerator has a static validParams, can we just get all these parameters from there?
then do:
params += DistributedRectilinearMeshGenerator::validParams();